File: 18_binary_tree_ii/06_identical_trees.py
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p is None:
            return q is None

        if q is None:
            return p is None

        return (
            p.val == q.val
            and self.isSameTree(p.left, q.left)
            and self.isSameTree(p.right, q.right)
        )

File: 18_binary_tree_ii/06_identical_trees.go
func isSameTree(p *TreeNode, q *TreeNode) bool {

	if p == nil {
		return q == nil
	}
	if q == nil {
		return p == nil
	}
	return p.Val == q.Val && isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
}